### DISCUSSION TITLE
Compile to CSS

### DISCUSSION THREAD

**Discussion by nattaylor** [winghongchan/dark-docs#24] [Q&A]
I'd like to inject this CSS into a `nativefier` app.

I tried some naive things like `lessc dark-docs-m3.user.css out.css` but I get compilation errors

Am I on the right track?

```
NameError: Error evaluating function `contrast`: variable @document-contrast is undefined in /private/tmp/dark.less on line 128, column 51:
127     --whiter-icons: brightness(325%);
128     --darkfilter: invert(100%) hue-rotate(180deg) contrast(@document-contrast);
129     --on: background-color 0.02s ease-in, border-color 0.06s ease, color 0.05s ease, filter 0.01s ease;
```

**Comment by winghongchan** [comment]
`@document-contrast` is undefined because that’s one of the parameters users can [adjust through the Stylus extension](https://github.com/openstyles/stylus/wiki/Usercss#how-do-i-customize-usercss). So unlike normal {less} variables, this one is defined on line 12 of [dark-docs-m3.user.css](https://github.com/winghongchan/dark-docs/blob/main/dark-docs-m3.user.css) in with the [UserCSS metadata](https://github.com/openstyles/stylus/wiki/Writing-UserCSS#var): 

```css
@var            number document-contrast "Document contrast while inverted" [76, 40, 100, 1, "%"]
```

I have never used `lessc` but you probably need to define `@document-contrast` yourself by adding a line like 

```diff
+   @document-contrast: 76%;
    --darkfilter: invert(100%) hue-rotate(180deg) contrast(@document-contrast);
```

(76% is the default value for `@document-contrast`)
